src/PrestaShopBundle/Controller/Admin/Sell/Order/CreditSlipController.php

```diff
@@ -0,0 +1,222 @@
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use PrestaShopBundle\Security\Annotation\DemoRestricted;
+use PrestaShopBundle\Service\Grid\ResponseBuilder;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Responsible for Sell > Orders > Credit slips page
+ */
+class CreditSlipController extends FrameworkBundleAdminController
+{
+    /**
+     * Show manufacturers listing page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     * @param CreditSlipFilters $creditSlipFilters
+     *
+     * @return Response
+     */
+    public function indexAction(
+        Request $request,
+        CreditSlipFilters $creditSlipFilters
+    ) {
+        $creditSlipGridFactory = $this->get('prestashop.core.grid.factory.credit_slip');
+        $creditSlipGrid = $creditSlipGridFactory->getGrid($creditSlipFilters);
+        $creditSlipOptionsForm = $this->getSlipOptionsFormHandler()->getForm();
+        $pdfByDateForm = $this->createForm(GeneratePdfByDateType::class, [], [
+            'method' => Request::METHOD_GET,
+        ]);
+        $pdfByDateForm->handleRequest($request);
+
+        return $this->render('@PrestaShop/Admin/Sell/Order/CreditSlip/index.html.twig', [
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'creditSlipGrid' => $this->presentGrid($creditSlipGrid),
+            'pdfByDateForm' => $pdfByDateForm->createView(),
+            'creditSlipOptionsForm' => $creditSlipOptionsForm->createView(),
+        ]);
+    }
+
+    /**
+     * Provides filters functionality
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request)
+    {
+        /** @var ResponseBuilder $responseBuilder */
+        $responseBuilder = $this->get('prestashop.bundle.grid.response_builder');
+
+        return $responseBuilder->buildSearchResponse(
+            $this->get('prestashop.core.grid.definition.factory.credit_slip'),
+            $request,
+            CreditSlipGridDefinitionFactory::GRID_ID,
+            'admin_credit_slips_index'
+        );
+    }
+
+    /**
+     * Generates PDF of requested credit slip by provided id
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param int $creditSlipId
+     *
+     * @return RedirectResponse
+     */
+    public function generatePdfAction($creditSlipId)
+    {
+        try {
+            $creditSlipId = new CreditSlipId((int) $creditSlipId);
+            die($this->get('prestashop.adapter.pdf.credit_slip_pdf_generator')->generatePDF([$creditSlipId]));
```

Comment:
No way 😄using `die()` is very bad because it ends the processing here. You cant hook later or finish some important processings required by Symfony after the controller response.

`return new Response($this->get('prestashop.adapter.pdf.credit_slip_pdf_generator')->generatePDF([$creditSlipId]));` does not work ?